test/navi/transform_test.clj

```diff
@@ -81,6 +87,37 @@
   (testing "nil"
     (is (= any? (p/transform nil)))))
 
+(deftest date-schema-transformations
+  (testing "DateSchema transforms to inst? predicate"
+    (let [schema (DateSchema.)]
+      (is (= inst? (p/transform schema)))))
+
+  (testing "DateTimeSchema transforms to inst? predicate"
+    (let [schema (DateTimeSchema.)]
+      (is (= inst? (p/transform schema)))))
+
+  (testing "inst? validates different date types"
```

Comment:
can we add a case for https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html ? this is what the `date-time` schema refers to actually